### PR TITLE
E2e test for sign-multichain

### PR DESCRIPTION
**Pull request by benmalcom**
This is the PR for sign-multichain e2e tests

### CODE REVIEW

File: packages/near-fast-auth-signer-e2e-tests/mpc-service.ts
Comment (Pessina) [inline packages/near-fast-auth-signer-e2e-tests/mpc-service.ts:1]: I see some parts of it are copied from the app source code. Can we import it instead of copy? So we don't need to maintain duplicated code.

File: packages/near-fast-auth-signer-e2e-tests/package.json
Comment (Pessina) [inline packages/near-fast-auth-signer-e2e-tests/package.json:20]: I'm not sure if it's possible. 

But can we avoid adding test dependencies on the app dependencies? In order to avoid a big package.

File: packages/near-fast-auth-signer-e2e-tests/tests/signMultiChain/sign-multi-chain.spec.ts
Comment (Pessina) [inline packages/near-fast-auth-signer-e2e-tests/tests/signMultiChain/sign-multi-chain.spec.ts:102]: Shall we create a ticket for this?

File: packages/near-fast-auth-signer-e2e-tests/firebase.ts
Comment (Pessina) [inline packages/near-fast-auth-signer-e2e-tests/firebase.ts:5]: I'm not sure, but I believe some of the functions for firebase are already create by @hcho112 . 

Like this, I think both do the same thing, no? https://github.com/near/fast-auth-signer/blob/b6bfb4516130c7de6798a5c3ec5b99d28b5d8fae/packages/near-fast-auth-signer-e2e-tests/utils/firebase.ts#L29

If that's the case, please re-use the ones from Danny and include yours on the same file.

**Review by Pessina — CHANGES_REQUESTED** [review]

File: packages/near-fast-auth-signer-e2e-tests/tests/signMultiChain/sign-multi-chain.spec.ts
Comment (benmalcom) [reply packages/near-fast-auth-signer-e2e-tests/tests/signMultiChain/sign-multi-chain.spec.ts:102]: Yes please.

File: packages/near-fast-auth-signer-e2e-tests/firebase.ts
Comment (benmalcom) [reply packages/near-fast-auth-signer-e2e-tests/firebase.ts:5]: Both were actually created by @hcho112, this PR has nothing to do with both files, I only rebased after @hcho112's merge to be update.

Both files are already in `main`.

File: packages/near-fast-auth-signer-e2e-tests/mpc-service.ts
Comment (benmalcom) [reply packages/near-fast-auth-signer-e2e-tests/mpc-service.ts]: Removed. 
It's actually useless now, it's already imported.

File: packages/near-fast-auth-signer-e2e-tests/firebase.ts
Comment (hcho112) [reply packages/near-fast-auth-signer-e2e-tests/firebase.ts:5]: @benmalcom if you look at latest main, you will notice that firebase.ts is moved under utils/firebase.ts

so this file is redundant. If you have added new functions, these should be allocated to utils/firebase.ts

File: packages/near-fast-auth-signer-e2e-tests/package.json
Comment (hcho112) [inline packages/near-fast-auth-signer-e2e-tests/package.json:17]: is extra space here intended?

File: packages/near-fast-auth-signer-e2e-tests/scripts/faucet.js
Comment (hcho112) [inline packages/near-fast-auth-signer-e2e-tests/scripts/faucet.js:5]: wonder if we should move this to secret for github action and for local dev, we should define them inside `.env`?

File: packages/near-fast-auth-signer-e2e-tests/scripts/faucet.js
Comment (hcho112) [inline packages/near-fast-auth-signer-e2e-tests/scripts/faucet.js:9]: there is exact same function exist on `ethers` on `near-fast-auth-signer/package.json` where:
```
const { ethers } = require('ethers');

function fromWei(wei) {
  return ethers.utils.formatEther(wei);
}
```
wonder if we should just import existing one?

File: packages/near-fast-auth-signer-e2e-tests/scripts/faucet.js
Comment (hcho112) [inline packages/near-fast-auth-signer-e2e-tests/scripts/faucet.js:26]: clean up if not needed

File: packages/near-fast-auth-signer-e2e-tests/testAccounts.json
Comment (hcho112) [inline packages/near-fast-auth-signer-e2e-tests/testAccounts.json:154]: these shouldn't be committed, it should be empty. 
I

File: packages/near-fast-auth-signer-e2e-tests/tests/signMultiChain/sign-multi-chain.spec.ts
Comment (hcho112) [inline packages/near-fast-auth-signer-e2e-tests/tests/signMultiChain/sign-multi-chain.spec.ts:58]: there is a util function called `isWalletSelectorLoaded` inside `walletSelector.ts` that can replace this

File: packages/near-fast-auth-signer-e2e-tests/utils/constants.ts
Comment (hcho112) [inline packages/near-fast-auth-signer-e2e-tests/utils/constants.ts:14]: clean up if not needed or leave a comment

File: packages/near-fast-auth-signer-e2e-tests/utils/multiChain.ts
Comment (hcho112) [inline packages/near-fast-auth-signer-e2e-tests/utils/multiChain.ts:8]: there is existing function on ethers with

```
export function toWei(eth: number): string {
  const wei = eth * 1e18;
  return wei.toString();
}
```

File: packages/near-fast-auth-signer-e2e-tests/utils/multiChain.ts
Comment (hcho112) [inline packages/near-fast-auth-signer-e2e-tests/utils/multiChain.ts:63]: delete?

File: packages/near-fast-auth-signer-e2e-tests/firebase.ts
Comment (benmalcom) [reply packages/near-fast-auth-signer-e2e-tests/firebase.ts]: @hcho112 you're right, I see it's useless now, i've removed it.

File: packages/near-fast-auth-signer-e2e-tests/scripts/faucet.js
Comment (benmalcom) [reply packages/near-fast-auth-signer-e2e-tests/scripts/faucet.js:6]: I'm indifferent about this.
But I think we should leave it here,  we can quickly see the use of the addresses from the comment on top of it instead of figuring out what it does an an `.env ` variable.

File: packages/near-fast-auth-signer-e2e-tests/scripts/faucet.js
Comment (Pessina) [inline packages/near-fast-auth-signer-e2e-tests/scripts/faucet.js:68]: Shouldn't we make the `address/chain` arguments of the function instead of hard code taht we will fetch only for ETH/BNB? 

So, in the future, will be easier to add test for more chains

File: packages/near-fast-auth-signer-e2e-tests/test-app/components/SignMultiChain.tsx
Comment (Pessina) [inline packages/near-fast-auth-signer-e2e-tests/test-app/components/SignMultiChain.tsx:14]: I think we are using `react-hook-form` and `yup` for form management on `fast-auth-signer`, no? 

If so, can you please re-use it here to make it consistent with the codebase?

File: packages/near-fast-auth-signer-e2e-tests/tests/sign/sign-transaction.spec.ts
Comment (Pessina) [inline packages/near-fast-auth-signer-e2e-tests/tests/sign/sign-transaction.spec.ts:119]: Plase explain the @ts-ignore

File: packages/near-fast-auth-signer-e2e-tests/tests/signMultiChain/sign-multi-chain.spec.ts
Comment (Pessina) [inline packages/near-fast-auth-signer-e2e-tests/tests/signMultiChain/sign-multi-chain.spec.ts:41]: Explain why this is necessary. This doesn't seems to be something that the user can do on his side while using the app

**Review by Pessina — CHANGES_REQUESTED** [review]

File: packages/near-fast-auth-signer-e2e-tests/scripts/faucet.js
Comment (benmalcom) [reply packages/near-fast-auth-signer-e2e-tests/scripts/faucet.js]: Done

File: packages/near-fast-auth-signer-e2e-tests/scripts/faucet.js
Comment (benmalcom) [reply packages/near-fast-auth-signer-e2e-tests/scripts/faucet.js]: Done

File: packages/near-fast-auth-signer-e2e-tests/test-app/components/SignMultiChain.tsx
Comment (benmalcom) [reply packages/near-fast-auth-signer-e2e-tests/test-app/components/SignMultiChain.tsx]: Done

File: packages/near-fast-auth-signer-e2e-tests/test-app/components/SignMultiChain.tsx
Comment (benmalcom) [reply packages/near-fast-auth-signer-e2e-tests/test-app/components/SignMultiChain.tsx]: Done

File: packages/near-fast-auth-signer-e2e-tests/tests/sign/sign-transaction.spec.ts
Comment (benmalcom) [reply packages/near-fast-auth-signer-e2e-tests/tests/sign/sign-transaction.spec.ts]: Removed.
It stops my tests from running as there's unimplemented with error `Property useLocalViewExecution is missing in type`, had to disable the warning before running mine, guess it's not so for every other person.

File: packages/near-fast-auth-signer-e2e-tests/scripts/faucet.js
Comment (benmalcom) [reply packages/near-fast-auth-signer-e2e-tests/scripts/faucet.js]: Done

File: packages/near-fast-auth-signer-e2e-tests/tests/signMultiChain/sign-multi-chain.spec.ts
Comment (benmalcom) [reply packages/near-fast-auth-signer-e2e-tests/tests/signMultiChain/sign-multi-chain.spec.ts:41]: `webpack-dev-server` adds a hidden iframe to projects (for hot reloading), that iframe stays on top of the page components and causes event propagation, which means a click event by playwright will have multiple effects, so it needs to be disbaled.

This can't happen in `prod` for ideal user because `webpack-dev-server` is used for development.

Comment (Pessina) [reply packages/near-fast-auth-signer-e2e-tests/tests/signMultiChain/sign-multi-chain.spec.ts:41]: Got it, please add this explanation on the code so it's easier for us to know why we are disabling it.

File: packages/near-fast-auth-signer-e2e-tests/tests/signMultiChain/sign-multi-chain.spec.ts
Comment (Pessina) [reply packages/near-fast-auth-signer-e2e-tests/tests/signMultiChain/sign-multi-chain.spec.ts:97]: Please add details to the ticket: https://github.com/near/fast-auth-signer/issues/259

File: packages/near-fast-auth-signer-e2e-tests/tests/signMultiChain/sign-multi-chain.spec.ts
Comment (benmalcom) [reply packages/near-fast-auth-signer-e2e-tests/tests/signMultiChain/sign-multi-chain.spec.ts]: Done

File: packages/near-fast-auth-signer-e2e-tests/tests/signMultiChain/sign-multi-chain.spec.ts
Comment (benmalcom) [reply packages/near-fast-auth-signer-e2e-tests/tests/signMultiChain/sign-multi-chain.spec.ts:100]: Done

**Review by Pessina — APPROVED** [review]

**Comment by Pessina** [comment]
@benmalcom thanks for fixing all the comments! LGTM!

**Comment by benmalcom** [comment]
@hcho112 can you approve this as the last change from [this PR](https://github.com/near/fast-auth-signer/pull/260) dismissed @Pessina review.

File: .github/workflows/playwright.yml
Comment (hcho112) [inline .github/workflows/playwright.yml:74]: why does it have to be passed down to .env instead of directly feed into process.env like 
`MULTICHAIN_TEST_ACCOUNT_ID: ${{ secrets.MULTICHAIN_TEST_ACCOUNT_ID }}`

**Review by hcho112 — APPROVED** [review]
Left one comment for clarification but rest looks good

File: .github/workflows/playwright.yml
Comment (benmalcom) [reply .github/workflows/playwright.yml]: @hcho112 that's right. 
Fixed.

**Review by hcho112 — APPROVED** [review]
LGTM

**Pull request merged**
